### PR TITLE
Ref 449

### PR DESCRIPTION
**Pull request by brettminnie**

### CODE REVIEW

File: roles/iam-kms/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+only get executed if the `kms_data` key is present in the meta-data
+
+####Example meta-data and output
+```yaml
+opg_data:
+  project: foo
+  environment: bar
+  
+kms_data:
+  - key_description: baz
```
Comment (carinadigital) [inline roles/iam-kms/README.md:13]: Isn't this a suffix?

File: roles/iam-kms/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+---
+
+- name: Set a date stamp for our key and key_alias
```
Comment (carinadigital) [inline roles/iam-kms/tasks/main.yml]: Why not move this to `create-update` where the variable is used?

File: roles/iam-kms/tasks/create-update.yml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  set_fact:
+    kms_tags:
+      - { "TagKey":"Environment","TagValue":"{{ opg_data.environment }}" }
+      - { "TagKey":"Stack","TagValue":"{{ target }}" }
+      - { "TagKey":"Project","TagValue":"{{ opg_data.project }}" }
+      - { "TagKey":"Role","TagValue":"{{ item.key_description }}" }
+      - { "TagKey":"Alias","TagValue":"{{ key_alias }}" }
+      - { "TagKey":"CreationDate","TagValue":"{{ date_stamp.stdout }}" }
+
+- name: Create our kms master key
```
Comment (carinadigital) [inline roles/iam-kms/tasks/create-update.yml:27]: Won't this create a master key every day in our non-prod stacks?

**Review by carinadigital — CHANGES_REQUESTED** [review]

File: roles/iam-kms/tasks/create-update.yml (hunk 3 above)
Comment (brettminnie) [reply roles/iam-kms/tasks/create-update.yml:27]: No, I have defined the meta-data only on the vpc. In the use case I have prod and non prod will have specific keys that are shared across the env for encrypt/decrypt, so it made more sense to define the meta-data only on the vpc, if the key doesn't exist in the meta, it skips the role entirely, also I have patched the role onto it's own playbook so there is no inter-dependancy with stack creation

File: roles/iam-kms/README.md (hunk 1 above)
Comment (brettminnie) [reply roles/iam-kms/README.md:13]: Possibly, but I use it in the tags and in the alias as well... bit of a hard one to define

File: roles/iam-kms/tasks/main.yml (hunk 2 above)
Comment (brettminnie) [reply roles/iam-kms/tasks/main.yml]: Will do

File: roles/iam-kms/README.md (hunk 1 above)
Comment (carinadigital) [reply roles/iam-kms/README.md:13]: Maybe just name the variable better to let us know it's a suffix, or just part of the naming scheme

**Pull request merged**